ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -192,37 +200,43 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 	}
 
-	// Check the current status of ray jobs before submitting.
-	jobInfo, err := rayDashboardClient.GetJobInfo(ctx, rayJobInstance.Status.JobId)
+	// Ensure k8s job has been created
+	jobName, err := r.CreateK8sJob(ctx, rayJobInstance)
 	if err != nil {
-		err = r.updateState(ctx, rayJobInstance, jobInfo, rayJobInstance.Status.JobStatus, rayv1alpha1.JobDeploymentStatusFailedToGetJobStatus, err)
-		// Dashboard service in head pod takes time to start, it's possible we get connection refused error.
-		// Requeue after few seconds to avoid continuous connection errors.
+		r.Log.Error(err, "failed to create k8s job")
+		err = r.updateState(ctx, rayJobInstance, nil, rayJobInstance.Status.JobStatus, rayv1alpha1.JobDeploymentStatusFailedJobDeploy, err)
```

Comment:
Calling `updateState` multiple times does not make sense to me. We can refer to the implementation of [ReplicaSet](https://sourcegraph.com/github.com/kubernetes/kubernetes@9e622d23648f4af4a12835de97484dc0e4859269/-/blob/pkg/controller/replicaset/replica_set.go?L710).

Reply:
Gotcha, I remember you made a similar improvement for RayService. 

I think we can implement your suggestion in a separate PR to keep the current PR smaller and self-contained, but let me know if you think otherwise.